django_lifecycle/mixins.py

```diff
@@ -140,42 +141,44 @@ def delete(self, *args, **kwargs):
         super().delete(*args, **kwargs)
         self._run_hooked_methods(AFTER_DELETE)
 
-    @cached_property
-    def _potentially_hooked_methods(self):
-        skip = set(get_unhookable_attribute_names(self))
+    @classmethod
+    def _potentially_hooked_methods(cls):
+        skip = set(cls._get_unhookable_attribute_names())
         collected = []
 
-        for name in dir(self):
+        for name in dir(cls):
             if name in skip:
                 continue
             try:
-                attr = getattr(self, name)
-                if ismethod(attr) and hasattr(attr, "_hooked"):
+                attr = getattr(cls, name)
+                if isfunction(attr) and hasattr(attr, "_hooked"):
```

Comment:
Minor semantic change right here - I think the type of the method/function under the hood depends on whether you're looking at it from the instance perspective (where methods are special) or from the class perspective (where methods are just functions and there's no magic going on).

I don't think it's problematic but I want to call it out for a little extra attention.